src/plugin/PluginBase.js

```diff
@@ -127,8 +120,7 @@ define([
      * @returns {string}
      */
     PluginBase.prototype.getName = function () {
-        throw new Error('implement this function in the derived class - getting type automatically is a bad idea,' +
-            'when the js scripts are minified names are useless.');
+        return this.pluginMetadata.name;
```

Comment:
if the default 'null' metada is not overridden, this cause an exception.
Do we intended to have this change of behavior? (or should we still send some nice error instead)